core/src/main/java/org/apache/carbondata/core/scan/filter/executer/IncludeFilterExecuterImpl.java

```diff
@@ -186,12 +314,60 @@ private boolean isScanRequired(byte[] blkMaxVal, byte[] blkMinVal, byte[][] filt
     return isScanRequired;
   }
 
+  private boolean isScanRequired(byte[] maxValue, byte[] minValue, byte[][] filterValue,
+      DataType dataType) {
+    for (int i = 0; i < filterValue.length; i++) {
+      if (filterValue[i].length == 0 || maxValue.length == 0 || minValue.length == 0) {
+        return isScanRequired(maxValue, minValue, filterValue);
+      } else {
+        switch (dataType) {
```

Comment:
Use existing methods of DataTypeUtil for conversions. And use comparator which is used in applyFilter method here